controller/common/manage_children.go

```diff
@@ -45,10 +45,45 @@ func ApplyUpdate(orig, update *unstructured.Unstructured) (*unstructured.Unstruc
 	if err != nil {
 		return nil, err
 	}
+	// Revert metadata fields that are known to be read-only, system fields.
+	// We should never try to update those, even if the user asks us to.
+	// See: https://github.com/GoogleCloudPlatform/metacontroller/issues/76
+	if err := revertObjectMetaSystemFields(newObj, orig); err != nil {
+		return nil, fmt.Errorf("failed to revert ObjectMeta system fields: %v", err)
+	}
 	dynamicapply.SetLastApplied(newObj, update.UnstructuredContent())
 	return newObj, nil
 }
 
+// objectMetaSystemFields is a list of JSON field names within ObjectMeta that
+// are both read-only and system-populated according to the comments in
+// k8s.io/apimachinery/pkg/apis/meta/v1/types.go.
+var objectMetaSystemFields = []string{
+	"selfLink",
+	"uid",
+	"resourceVersion",
+	"generation",
+	"creationTimestamp",
+	"deletionTimestamp",
+}
+
+// revertObjectMetaSystemFields overwrites the read-only, system-populated
+// fields of ObjectMeta in newObj to match what they were in orig.
+func revertObjectMetaSystemFields(newObj, orig *unstructured.Unstructured) error {
+	for _, fieldName := range objectMetaSystemFields {
+		val, found, err := unstructured.NestedFieldNoCopy(orig.UnstructuredContent(), "metadata", fieldName)
+		if err != nil {
+			return err
+		}
+		if found {
+			unstructured.SetNestedField(newObj.UnstructuredContent(), val, "metadata", fieldName)
+		} else {
+			unstructured.RemoveNestedField(newObj.UnstructuredContent(), "metadata", fieldName)
```

Comment:
To verify, we omit a field if we cannot find it via `NestedFieldNoCopy`. This means that user adds the read-only field.